### PR TITLE
feat: Introduce payload primitives

### PR DESCRIPTION
**Pull request by ghost**
Continue #8369 
Ref: #8330

### CODE REVIEW

File: crates/payload/primitives/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+/// Payload types
+pub trait PayloadTypes: DeserializeOwned + Serialize + Debug + Unpin + Send + Sync + Clone {
+    /// The RPC payload attributes type the CL node emits via the engine API.
+    type PayloadAttributes: PayloadAttributes + Unpin;
+
+    /// The payload attributes type that contains information about a running payload job.
+    type PayloadBuilderAttributes: PayloadBuilderAttributes<RpcPayloadAttributes = Self::PayloadAttributes>
+        + Clone
+        + Unpin;
+}
```
Comment (mattsse) [inline crates/payload/primitives/src/lib.rs:35]: can we remove this for now,

we can try to introduce this in a follow-up

**Review by mattsse — CHANGES_REQUESTED** [review]
I like this restructuring.

in order to complete the original idea, we'd like to rollout the payloadtypes trait.

I'd like to keep this pr limited to the introduction of the primitives crate and move away from engine types in a followup pr

File: crates/payload/primitives/src/lib.rs (hunk 1 above)
Comment (0xqd) [reply crates/payload/primitives/src/lib.rs:35]: yes exactly, I'm stuck at this as well, since I don't have a good design to follow with.

**Review by mattsse — APPROVED** [review]
lgtm, nextup Payloadtypes

**Pull request merged**